[#265] Update report generation test

iperdomo opened the pull request:
Update the end-to-end tests by:

- Mocking AWS S3 with an static http server, serving a generated survey zip file
- Use `appId` and fallback to `uploadUrl` when possible to access the configuration map
- Removes `:exclusions` because Jackson databind library is used by the report generation code
- Fix the configuration with an extra `'` in a `/reports/` endpoint

iperdomo committed:
[#265] Update report generation test

dlebrero commented on backend/test/akvo/flow_services/end_to_end_test.clj:
```diff
@@ -213,7 +213,7 @@
 
 (deftest error-in-report-generation
-  (let [survey-id (System/currentTimeMillis)
+  (let [survey-id 43993002
```
If I remember correctly, we want to change the survey-id between runs so that we know that each run generates a new report, because flow-services keeps some cache, of which survey-id is part of the key.

If this is the case, maybe the test can copy the pregenerated zip file to the correct place in the s3 mock.

stellanl replied:
Simplest  approach is to begin this test by requesting services to invalidate the cache.

Otherwise, to be a correct published form, surveyId is both supposed to be
1. _in_ the xml file
2. the name of the xml file in the zip archive.

But the report generator never actually checks 1. so some renaming would be enough.

dlebrero reviewed (commented):
What is the relationship of the content of the zip file and the generated reports? Can we remove some of the test setup logic because it is not longer needed?

stellanl reviewed (commented):
I don't understand the test enough to tell if it would be good to begin this test by requesting services to invalidate the cache for this (now fixed) surveyId. Could there be an old report file cached?

iperdomo committed:
[#265] Update travis settings + build script

iperdomo committed:
[#265] Update Flow dependency + Invalidate cache between tests

I have tried to copy a "sample" file and use System/currentTimemillis,
but somehow the Java code that opens the connection and reads the XML
was failing to parse the copy with a very cryptic error:

    Unexpected EOF in prolog at [row,col {unknown-source}]: [1,0]

It suggests that the file is corrupted or something similar. When
verifying the integrity of the copy (with `sha256sum`) the
resulting file was "identical".

As @stellanl suggested we now issue a POST request to `/invalidate` to
avoid using a cached report between tests.

dlebrero commented on backend/project.clj:
```diff
@@ -34,7 +34,7 @@
                  [org.xerial/sqlite-jdbc "3.7.2"]
                  ; Akvo FLOW dependencies
-                 [org.akvo.flow/akvo-flow "v1.9.49-211-g94e20820" :classifier "classes"]
+                 [org.akvo.flow/akvo-flow "v1.9.49-193-g855fa8e4" :classifier "classes"]
```
Are you sure about this? It doesn't seem right to change to an older version.

iperdomo replied:
Let's wait for @stellanl to comment on this ... My understanding is that this revision is 193 commits _ahead_ of the tag `v1.9.49`. That's why I asked about the relationship between `master` and `develop` in akvo-flow repo.

stellanl replied:
When I made the edit 1.10 was not yet tagged.

stellanl committed:
[#265]Use latest Flow classes.
